packages/react/src/utils/useComponentRenderer.ts

```diff
@@ -61,12 +65,16 @@ export function useComponentRenderer<
     propGetter = (props) => props,
     extraProps,
     customStyleHookMapping,
+    skipGeneratingStyleHooks = false,
   } = settings;
 
   const className = resolveClassName(classNameProp, state);
   const styleHooks = React.useMemo(() => {
+    if (skipGeneratingStyleHooks) {
+      return {};
```

Comment:
`{}` can be defined outsite of the hook so it's not recreated every time it runs.